lib/datadog/di/serializer.rb

```diff
@@ -155,7 +158,8 @@ def serialize_value(value, name: nil,
             end
           end
 
-          serialized = {type: class_name(cls)}
+          # steep assuming invalid types
+          serialized = {}.update(type: class_name(cls))
```

Comment:
Hmm, this code is now worse, in order to please steep. Can we add a steep ignore instead?